eZ/Publish/API/Repository/Tests/LocationServiceTest.php

```diff
@@ -2520,6 +2520,49 @@ public function testCopySubtreeIncrementsChildCountOfNewParent()
         $this->assertEquals($childCountBefore + 1, $childCountAfter);
     }
 
+    /**
+     * @covers \eZ\Publish\API\Repository\LocationService::copySubtree()
+     */
+    public function testCopySubtreeWithInvisibleChild(): void
+    {
+        $repository = $this->getRepository();
+        $locationService = $repository->getLocationService();
+
+        // Hide child Location
+        $locationService->hideLocation($locationService->loadLocation($this->generateId('location', 53)));
+
+        $this->refreshSearch($repository);
+
+        $locationToCopy = $locationService->loadLocation($this->generateId('location', 43));
+
+        $expected = $this->loadSubtreeProperties($locationToCopy);
+
+        $mediaLocationId = $this->generateId('location', 43);
+        $demoDesignLocationId = $this->generateId('location', 56);
+        $locationService = $repository->getLocationService();
+
+        $locationToCopy = $locationService->loadLocation($mediaLocationId);
+
+        $newParentLocation = $locationService->loadLocation($demoDesignLocationId);
+
+        $copiedLocation = $locationService->copySubtree(
+            $locationToCopy,
+            $newParentLocation
+        );
+
+        $this->refreshSearch($repository);
+
+        // Load Subtree properties after copy
+        $actual = $this->loadSubtreeProperties($copiedLocation);
+
+        self::assertEquals(count($expected), count($actual));
+
+        foreach ($actual as $key => $properties) {
+            self::assertEquals($expected[$key]['hidden'], $properties['hidden']);
+            self::assertEquals($expected[$key]['invisible'], $properties['invisible']);
+        }
```

Comment:
In general usage of `self::assertSame` is preferred. It's similar to using `==` vs `===` - if there is no reason for non-strict comparison, strict should be used.